packages/input/src/resources.ts

```diff
@@ -7,26 +7,36 @@ export class InputStruct {
   /**
    * Whether to enable pointer lock on the canvas.
    */
-  @struct.bool declare enablePointerLock: boolean;
+  enablePointerLock: boolean = false;
 
   /**
    * Whether the pointer is currently locked.
    */
-  @struct.bool declare isPointerLocked: boolean;
+  isPointerLocked: boolean = false;
 
   /**
    * Whether the pointer is currently down.
    */
-  @struct.bool declare isPointerDown: boolean;
+  isPointerDown: boolean = false;
 
   /**
    * The currently pressed keys.
    * Each index is a `Key` enum value.
    * The value is `1` if the key is pressed, `0` otherwise.
    */
-  @struct.array({ length: NUM_KEYS, type: "u8" }) declare keys: Uint8Array;
+  #keys: u64 = 0n;
```

Comment:
I opted to handle this differently to keep the type from being massive and to reduce the memory footprint of it (not that it matters too much for resources), obviously feel free to handle this however you'd like 😃